rewrite-java-test/src/test/java/org/openrewrite/java/RemoveObjectsIsNullTest.java

```diff
@@ -63,58 +62,35 @@ public void test() {
 
     @Issue("https://github.com/openrewrite/rewrite/issues/1547")
     @Test
-    void transformCallToIsNullNeedsParentheses() {
+    void downcastToPrimitiveBoolean() {
         rewriteRun(
           java(
             """
               import static java.util.Objects.isNull;
               public class A {
                   public void test() {
-                      boolean a = true, b = false;
+                      Boolean a = true, b = false;
                       if (isNull(a || b)) {
                           System.out.println("a || b is null");
                       }
                   }
               }
-              """,
-            """
-              import static java.util.Objects.isNull;
-              public class A {
-                  public void test() {
-                      boolean a = true, b = false;
-                      if ((a || b) == null) {
-                          System.out.println("a || b is null");
-                      }
-                  }
-              }
               """
           )
         );
     }
 
-    @Issue("https://github.com/openrewrite/rewrite/issues/1547")
+    @Issue("https://github.com/openrewrite/rewrite/issues/3915")
     @Test
-    void transformCallToNonNull() {
+    void upcastToBoolean() {
         rewriteRun(
           java(
             """
-              import static java.util.Objects.nonNull;
-              public class A {
-                  public void test() {
-                      boolean a = true;
-                      if (java.util.Objects.nonNull(a)) {
-                          System.out.println("a is non-null");
-                      }
-                  }
-              }
-              """,
-            """
-              import static java.util.Objects.nonNull;
+              import static java.util.Objects.isNull;
               public class A {
-                  public void test() {
-                      boolean a = true;
-                      if (a != null) {
-                          System.out.println("a is non-null");
+                  public void test(boolean a) {
+                      if (isNull(a)) {
```

Comment:
This should be changed, shouldn't it?

Reply:
I had my doubts about this case; when we upcast from `boolean` to `Boolean` it doesn't really make sense to even call `isNull` with an upcast to `Boolean` for that method call. Perhaps we should completely remove the method call in those cases?

Reply:
Oh, yeah! And replace !isNull by `false` 😅

Reply:
Yup; I've changed the logic now to add simple `true` or `false` instead. [In rewrite-launchdarkly](https://github.com/openrewrite/rewrite-launchdarkly/blob/87279dabe8a7d57260575381f4da40dd2262f251/src/main/java/org/openrewrite/launchdarkly/RemoveBoolVariation.java#L67) we follow such a replacement up with `SimplifyConstantIfBranchExecution`, but that would require this recipe to move to rewrite-static-analysis. 🤔